NixOS/nixpkgs
# patchutils: 0.3.3 -> 0.3.4, keep 0.3.3 for fetchpatch

###### Motivation for this change

See #38618

Closes #38618

This upgrades patchutils from 0.3.3 -> 0.3.4. However, there were issues with its usage from `fetchpatch` combined with patchutils 0.3.4. One of the possible solutions that was raised by @dezgeg was to add both versions of patchutils. That is what I have attempted to to in this PR. 

To test whether `fetchpatch` still works:

```
nix-build -A pkgs.clightning.patches .
```

Tried the following to see whether both versions of patchutils are working:

```
nix-build -A pkgs.patchutils_0_3_3 .
result/bin/lsdiff
nix-build -A pkgs.patchutils_0_3_4 .
result/bin/lsdiff
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Assured whether relevant documentation is up to date

---
